data/src/test/java/org/oppia/android/data/backends/gae/RemoteAuthNetworkInterceptorTest.kt

```diff
@@ -0,0 +1,207 @@
+import dagger.BindsInstance
+import dagger.Component
+import dagger.Module
+import dagger.Provides
+import okhttp3.Headers
+import okhttp3.OkHttpClient
+import okhttp3.mockwebserver.MockResponse
+import okhttp3.mockwebserver.MockWebServer
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.oppia.android.data.backends.gae.api.TopicService
+import org.oppia.android.testing.RobolectricModule
+import org.oppia.android.testing.TestDispatcherModule
+import org.oppia.android.testing.TestLogReportingModule
+import org.oppia.android.testing.network.MockTopicService
+import org.oppia.android.util.data.DataProvidersInjector
+import org.oppia.android.util.data.DataProvidersInjectorProvider
+import org.robolectric.Shadows
+import org.robolectric.annotation.Config
+import org.robolectric.annotation.LooperMode
+import retrofit2.Retrofit
+import retrofit2.converter.moshi.MoshiConverterFactory
+import java.util.concurrent.TimeUnit
+import javax.inject.Inject
+import javax.inject.Qualifier
+import javax.inject.Singleton
+
+/** Tests for [RemoteAuthNetworkInterceptor] */
+@RunWith(AndroidJUnit4::class)
+@Config(application = RemoteAuthNetworkInterceptorTest.TestApplication::class)
+@LooperMode(LooperMode.Mode.PAUSED)
+class RemoteAuthNetworkInterceptorTest {
+
+  @Inject
+  lateinit var remoteAuthNetworkInterceptor: RemoteAuthNetworkInterceptor
+
+  @Inject
+  lateinit var context: Context
+
+  private lateinit var retrofit: Retrofit
+
+  lateinit var topicService: TopicService
+
+  private val testVersionName = "1.0"
+
+  private val testVersionCode = 1
+
+  private val topicName = "Topic1"
+
+  private val mockWebServer = MockWebServer()
+
+  @Before
+  fun setUp() {
+    setUpTestApplicationComponent()
+    setUpApplicationForContext()
+    setUpRetrofit()
+  }
+
+  private fun setUpApplicationForContext() {
+    val packageManager = Shadows.shadowOf(context.packageManager)
+    val applicationInfo =
+      ApplicationInfoBuilder.newBuilder()
+        .setPackageName(context.packageName)
+        .build()
+    val packageInfo =
+      PackageInfoBuilder.newBuilder()
+        .setPackageName(context.packageName)
+        .setApplicationInfo(applicationInfo)
+        .build()
+    packageInfo.versionName = testVersionName
+    packageInfo.versionCode = testVersionCode
+    packageManager.installPackage(packageInfo)
+  }
+
+  @Test
+  fun testNetworkInterceptor_withoutHeaders_addsCorrectHeaders() {
+    mockWebServer.enqueue(MockResponse().setBody("{}"))
+    val call = topicService.getTopicByName(topicName)
+    val serviceRequest = call.request()
+    assertThat(serviceRequest.header("api_key")).isNull()
+    assertThat(serviceRequest.header("app_package_name")).isNull()
+    assertThat(serviceRequest.header("app_version_name")).isNull()
+    assertThat(serviceRequest.header("app_version_code")).isNull()
+
+    call.execute()
+    val interceptedRequest = mockWebServer.takeRequest(timeout = 8, unit = TimeUnit.SECONDS)
```

Comment:
Suggest going with 10 seconds for parity with TestCoroutineDispatcher (in fact you could use the fields from there for the time & unit if you'd prefer).

Reply:
Done